### ISSUE TITLE
Detect redundant `Generic` in a class's bases?

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#427]
For example:

```py
from typing import Generic, TypeVar

T = TypeVar("T")

class Foo(Generic[T]): ...
class Bar(Foo[T], Generic[T]): ...
```

The multiple inheritance with `Generic` is not required for `Bar`; the class is already generic because it inherits from `Foo`. You could just do this instead, and it's arguably cleaner:

```py
from typing import Generic, TypeVar

T = TypeVar("T")

class Foo(Generic[T]): ...
class Bar(Foo[T]): ...
```

This might be pretty tricky to implement; I'm not sure. The hypothetical new check should not complain about any of these:

```py
from typing import Generic, TypeVar

T = TypeVar("T")
U = TypeVar("U")

class Eggs: ...
class Foo(Generic[T]): ...
class Bar(Foo[Eggs], Generic[T]): ...
class Baz(Foo[T], Generic[T, U]): ...
```

And there are edge cases such as this, where explicitly inheriting from `Generic` is not strictly required, but you could argue that it improves readability (as you can see at a glance all the TypeVars together):

```py
from typing import Generic, TypeVar

T1 = TypeVar("T1")
T2 = TypeVar("T2")

class A(Generic[T1]): ...
class B(Generic[T1]): ...
class C(Generic[T1]): ...
class D(Generic[T1]): ...

class E(A[int], B[T1], C[str], D[T2], Generic[T1, T2]): ...
```

This idea was mentioned by @JelleZijlstra in https://github.com/python/typeshed/pull/10610#discussion_r1304371701

**Comment by Avasam** [comment]
I'm for it. I think it could actually help newer type contributors (in or outside typeshed) with their understanding of `Generic` when it's flagged as redundant.
